File: Npc/11000504.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11000504: Rue
/// </summary>
public class _11000504 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:0831180407002187$
    // - Can I help you?
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:1217012607005230$
                // - This place is... How do I say? Full of contradictions. A rich, gorgeous facade that hides a lingering melancholy. This hotel is luxurious, but it's not for the regular people who live around it. Those people belong to a different world.
                return -1;
        }

        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11000504.py
""" 11000504: Rue """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0831180407002187$
        # - Can I help you?
        return None # TODO

    def __30(self, pick: int) -> int:
        # $script:1217012607005230$
        # - This place is... How do I say? Full of contradictions. A rich, gorgeous facade that hides a lingering melancholy. This hotel is luxurious, but it's not for the regular people who live around it. Those people belong to a different world.
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.CLOSE
        return Option.NONE
